sgx-types/src/ssa.rs

```diff
@@ -8,62 +8,36 @@
 
 use bitflags::bitflags;
 use core::{
-    mem::{align_of, size_of},
+    mem::{align_of, size_of, MaybeUninit},
     num::NonZeroU32,
 };
+use enumerate::enumerate;
 use intel_types::*;
 #[cfg(test)]
 use testing::testaso;
 
-/// Section 38.9.1.1, Table 38-10
-#[derive(Copy, Clone, Debug)]
-pub enum Exception {
-    /// Divider exception.
-    Divider,
+enumerate! {
+    /// Section 38.9.1.1, Table 38-9
+    #[derive(Copy, Clone)]
+    pub enum ExitType: u8 {
+        /// Hardware
+        Hardware = 0b011,
 
-    /// Debug exception.
-    Debug,
-
-    /// Breakpoint exception.
-    Breakpoint,
-
-    /// Bound range exceeded exception.
-    BoundRange,
-
-    /// Invalid opcode exception.
-    InvalidOpCode,
-
-    /// General protection exception. Only reported if SECS.MISCSELECT.EXINFO = 1.
-    GeneralProtection,
-
-    /// Page fault exception. Only reported if SECS.MISCSELECT.EXINFO = 1.
-    PageFault,
-
-    /// x87 FPU floating-point error.
-    FloatingPoint,
-
-    /// Alignment check exceptions.
-    AlignmentCheck,
-
-    /// SIMD floating-point exceptions.
-    Simd,
+        /// Software
+        Software = 0b110,
+    }
 }
 
 /// Section 38.9.1.1, Table 38-9
-#[derive(Copy, Clone, Debug)]
-pub enum ExitType {
-    /// Hardware exit.
-    Hardware,
-
-    /// Software exit.
-    Software,
+#[repr(C, align(4))]
```

Comment:
Could use some documentation for this struct.

Reply:
I'm not sure what you're looking for here. This isn't a new struct. It is just a rework of the existing one. I would like better documentation here, but I'm still not 100% clear on how it is used yet. However, we're now at the place in the SGX code where we can start actually handling these errors. So we'll know shortly.

Reply:
I was just thinking about it because I had to consult the documentation about the `valid` field. But I can create an issue for this and do it as a separate PR.